frontend/src/pages/auth/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,16 +1,97 @@
-import { Button } from '~/libs/components/components.js';
+import { UserValidationMessage } from 'shared/build/packages/users/users.js';
+
+import { Button, Icon, Input, Link } from '~/libs/components/components.js';
+import { AppRoute } from '~/libs/enums/app-route.enum';
+import { useAppForm, useCallback, useState } from '~/libs/hooks/hooks.js';
+import {
+  type UserSignInRequestDto,
+  userSignInValidationSchema,
+} from '~/packages/users/users.js';
+
+import { DEFAULT_LOGIN_PAYLOAD } from './libs/constants/constants.js';
+import styles from './styles.module.scss';
 
 type Properties = {
-  onSubmit: () => void;
+  onSubmit: (payload: UserSignInRequestDto) => void;
 };
 
-const SignInForm: React.FC<Properties> = () => (
-  <>
-    <h1>Sign In</h1>
-    <form>
-      <Button label="Sign in" />
-    </form>
-  </>
-);
+const SignInForm: React.FC<Properties> = ({ onSubmit }) => {
+  const { control, errors, handleSubmit } = useAppForm({
+    defaultValues: DEFAULT_LOGIN_PAYLOAD,
+    validationSchema: userSignInValidationSchema,
+  });
+  const [isPasswordVisible, setPasswordVisibility] = useState(false);
+
+  const handleTogglePasswordVisibility = useCallback((): void => {
+    setPasswordVisibility((previousValue) => !previousValue);
+  }, [setPasswordVisibility]);
+
+  const handleFormSubmit = useCallback(
+    (event_: React.BaseSyntheticEvent): void => {
+      void handleSubmit(onSubmit)(event_);
+    },
+    [handleSubmit, onSubmit],
+  );
+
+  return (
+    <div className={styles.wrapper}>
+      <h2 className={styles.title}>Hello!</h2>
+      <form
+        className={styles.form}
+        name="loginForm"
+        onSubmit={handleFormSubmit}
+      >
+        <fieldset className={styles.fieldset}>
+          <Input
+            name="email"
+            type="email"
+            placeholder="Enter your email"
+            label="Email"
+            control={control}
+            errors={errors}
+            className={
+              errors?.email?.message === UserValidationMessage.EMAIL_REQUIRE
+                ? styles.errorInput
+                : ''
```

Comment:
This should be done inside Input component - it's already checking if field has errors - so just place styling there